pkg/build/api/types.go

```diff
@@ -707,6 +771,28 @@ type BuildRequest struct {
 
 	// Env contains additional environment variables you want to pass into a builder container
 	Env []kapi.EnvVar
+
+	// BuildCauses can pass information from the controllers to the buildconfig
+	// generator about who/what has triggered a build through the build request.
+	BuildCauses []*BuildTriggerCauses
```

Comment:
How is this different from the Info field you're adding to the BuildSpec? Why do we seem to be adding similar info in both places?